gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/log/LogElasticsearchRepository.java

```
@@ -132,4 +138,24 @@ public LogResponse<AggregatedMessageLog> searchAggregatedMessageLog(QueryContext
             )
             .blockingGet();
     }
+
+    private String getQueryIndexesFromDefinitionVersions(QueryContext queryContext, List<DefinitionVersion> definitionVersions) {
+        var clusters = ClusterUtils.extractClusterIndexPrefixes(configuration);
+        var indexV2Request = this.indexNameGenerator.getWildcardIndexName(queryContext.placeholder(), Type.REQUEST, clusters);
+        var indexV4Metrics = this.indexNameGenerator.getWildcardIndexName(queryContext.placeholder(), Type.V4_METRICS, clusters);
+
+        var indexes = new ArrayList<String>();
+
+        if (definitionVersions == null || definitionVersions.isEmpty()) {
+            indexes.add(indexV4Metrics);
+        } else {
```

I am wondering if we should handle the case where definitionVersions is null or empty... 
I don't know if I prefer to throw an error or to search in both index 🙈

---

I would say both by default no if definitionVersions is null or empty 🤷🏼

---

I would say that if it's null or empty we could search in both indexes 🤷🏼

---

I think that makes the most sense for me, too. I will add that in 👍